core/src/main/java/hudson/util/RobustReflectionConverter.java

```diff
@@ -451,6 +451,10 @@ private boolean fieldDefinedInClass(Object result, String attrName) {
 
     protected Object unmarshalField(final UnmarshallingContext context, final Object result, Class type, Field field) {
         Converter converter = mapper.getLocalConverter(field.getDeclaringClass(), field.getName());
+        // TODO: Is there a better way to make something like this work? Does it break any custom converters?
+        if (converter == null && new RobustCollectionConverter(mapper, reflectionProvider).canConvert(type)) {
+            converter = new RobustCollectionConverter(mapper, reflectionProvider, field.getGenericType());
+        }
```

Comment:
In particular I am wondering about things like [NodeList](https://github.com/jenkinsci/jenkins/blob/2d2adefabdf9edf025a277c95d1fc9d33bbfe14c/core/src/main/java/hudson/slaves/NodeList.java) and [AxisList](https://github.com/jenkinsci/matrix-project-plugin/blob/7ef205d818d6386125e0c042dd9093a7bbe883cd/src/main/java/hudson/matrix/AxisList.java) which extend `ArrayList`. They might be ok because of the exact equality comparisons [here](https://github.com/x-stream/xstream/blob/8472df83122b6b0840fc6d8c7945b5b55c25fb9d/xstream/src/java/com/thoughtworks/xstream/converters/collections/CollectionConverter.java#L64-L74), but I am not sure. I guess also that those classes don't support any of the desired recoverability features in `RobustCollectionConverter`, but that shouldn't be affected by my PR.

Reply:
I confirmed that `NodeList` is unaffected by this change - its custom converter continues to be used and its deserialization is not robust against invalid values.

Reply:
Looking at https://github.com/jenkinsci/jenkins/pull/342, I think we only expect `getLocalConverter` to matter for fields that have the `@XStreamConverter` annotation, which seems to be more or less unused based on [this search](https://github.com/search?type=code&q=+owner%3Ajenkinsci+%22%40XStreamConverter%22+-path%3Asrc%2Ftest+-path%3Axstream-distribution), and also none of those annotations are for collections or maps.